### PR TITLE
chore(react-styled-system): fixed linter errors

### PR DESCRIPTION
**Pull request by evwilkin**
**What**: Towards #3592 - resolves existing linter issues for react-styled-system

### CODE REVIEW

File: packages/patternfly-4/react-styled-system/scripts/copyTS.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
 const path = require('path');
 const glob = require('glob');
 const fse = require('fs-extra');
+/* eslint-enable @typescript-eslint/no-var-requires */
```
Comment (dlabrecq) [inline packages/patternfly-4/react-styled-system/scripts/copyTS.js]: This is a build script. Not sure we need to apply linter here? Perhaps a different config for js?

File: packages/patternfly-4/react-styled-system/src/components/StyledSystem/StyledTheme.d.ts [hunk 2]
```diff
@@ -161,7 +162,7 @@ export interface StyledConstants {
 }
 
 export interface PatternFlyThemeProviderProps extends HTMLProps<HTMLDivElement> {
-  theme?: Object;
+  theme?: Record<string, any>;
```
Comment (dlabrecq) [inline packages/patternfly-4/react-styled-system/src/components/StyledSystem/StyledTheme.d.ts]: Do we want to change the type or wait for a breaking change release?

File: packages/patternfly-4/react-styled-system/src/components/StyledSystem/StyledTheme.d.ts [hunk 3]
```diff
@@ -161,6 +162,7 @@ export interface StyledConstants {
 }
 
 export interface PatternFlyThemeProviderProps extends HTMLProps<HTMLDivElement> {
+  // eslint-disable-next-line @typescript-eslint/ban-types
```
Comment (seanforyou23) [inline packages/patternfly-4/react-styled-system/src/components/StyledSystem/StyledTheme.d.ts]: Do you know why we need the `Object` type here? If we can safely remove the general type in favor of the primitive `object` I think that's the way to go, then we don't need the ignore eslint statement

File: packages/patternfly-4/react-styled-system/src/components/StyledSystem/StyledTheme.d.ts [hunk 4]
```diff
@@ -161,7 +162,7 @@ export interface StyledConstants {
 }
 
 export interface PatternFlyThemeProviderProps extends HTMLProps<HTMLDivElement> {
-  theme?: Object;
+  theme?: object;
```
Comment (dlabrecq) [inline packages/patternfly-4/react-styled-system/src/components/StyledSystem/StyledTheme.d.ts]: Shouldn't this be `Object` not lowecase

Comment (evwilkin) [reply packages/patternfly-4/react-styled-system/src/components/StyledSystem/StyledTheme.d.ts]: @seanforyou23 pointed out [above](https://github.com/patternfly/patternfly-react/pull/3719#discussion_r379619119) that typically [Object is used incorrectly and object is preferred](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#general-types)

File: packages/patternfly-4/react-styled-system/src/components/StyledSystem/StyledTheme.d.ts (hunk 3 above)
Comment (evwilkin) [reply packages/patternfly-4/react-styled-system/src/components/StyledSystem/StyledTheme.d.ts]: Updated based on [https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#general-types](feedback)

**Pull request merged**